src/game-master/middlewares/phoenix-gm-middleware.js

```diff
@@ -54,6 +54,10 @@ function send(store, payload) {
 function handleServerMessage(store, message) {
     let action;
 
+    if (MESSAGE_NAME.gameMasterSessionState && message.puzzleIndex === null) {
```

Comment:
I do remember a lot of concerns about it. And finally in _qd-ui it was transformed to
```
currentRoundIndex: typeof puzzleIndex === 'number' ? puzzleIndex : -1,
```